xl_auth/public/views.py

```diff
@@ -43,10 +43,54 @@ def home():
             return redirect(redirect_url)
         else:
             flash_errors(login_form)
+
     return render_template('public/home.html', login_form=login_form,
                            next_redirect_url=request.args.get('next'))
 
 
+@blueprint.route('/forgot_password/', methods=['GET', 'POST'])
+def forgot_password():
+    """Forgot password form."""
+    forgot_password_form = ForgotPasswordForm(request.form)
+    if request.method == 'POST':
+        if forgot_password_form.validate_on_submit():
+            user = User.get_by_email(forgot_password_form.username.data)
```

Comment:
What happens if there is no user?

Reply:
Sorry, part of the previous PR. This block wouldn't get executed unless there was indeed a user. Validated by the *ForgotPasswordForm*:

```python
class ForgotPasswordForm(FlaskForm):
    """Reset password form."""

    username = StringField(_('Email'), validators=[DataRequired()])

    def validate(self):
        """Validate the form."""
        initial_validation = super(ForgotPasswordForm, self).validate()

        if not initial_validation:
            return False

        user = User.get_by_email(self.username.data)
        if user:
            return True
        else:
            self.username.errors.append(_('Unknown username/email'))
            return False
```